File: tests/rspec/pages/utilities/translate_edit.rb
class TranslateEdit < ControlPanelPage
  set_url_matcher /utilities\/translate\/\w+\/edit/

  element :heading, 'div.w-12 form h1'

  elements :items, 'form fieldset'

  element :submit_button, 'form div.form-btns.form-btns-top input[type="submit"]'

  def load
    self.open_dev_menu
    click_link 'Utilities'
    click_link 'English'
    self.all('ul.toolbar li.edit a')[0].click # The addons_lang.php edit link
  end
end

File: tests/cypress/cypress/elements/pages/utilities/TranslateEdit.js
import ControlPanel from '../ControlPanel'

class TranslateEdit extends ControlPanel {
  constructor() {
    super()
    this.urlMatcher = /utilities\/translate\/\w+\/edit/

    this.elements({

      'heading': '.ee-main .title-bar .title-bar__title',

      'items': 'form fieldset',

      'submit_button': 'form .tab-bar__right-buttons .form-btns input[type="submit"]'
    })
  }

  load() {
    this.open_dev_menu()
    cy.get('body > .dropdown').find('a:contains("Utilities")').click()
    this.get('wrap').find('a:contains("CP Translation")').click()
    this.get('wrap').find('a:contains("English")').click()
    this.get('wrap').find('.toolbar a.edit').first().click() // The addons_lang.php edit link
  }

}
export default TranslateEdit;